File: whatsapp#/python/send_file.py
import requests

domain = "https://wa-XXX.4jawaly.com/"
params = {
    "token": "XXXXXXXXX",
    "phone": "19292439373",
    "body": "https://cdn-dev.4jawaly.com/spam-regulation.pdf",
    "quotedMsgId": "",
    "filename": "filename.pdf"
}

if not params["token"]:
    print("token: It must not be a null value")
    exit()

if not params["phone"]:
    print("phone: It must not be a null value")
    exit()

if not params["body"]:
    print("body: It must not be a null value")
    exit()

url = f"{domain}api/v1/message/file"
headers = {"Accept": "application/json"}

response = requests.post(url, headers=headers, data=params)
result = response.json()

code = result["code"]
message = "تم الارسال بنجاح" if response.status_code == 200 else result["message"]
id = result["id"]

print(f"code: {code}")
print(f"message: {message}")
print(f"id: {id}")

File: whatsapp#/nodejs/send_file.js
const axios = require('axios');

const domain = "https://wa-XXX.4jawaly.com/";
const params = {
    token: "XXXXXXXXX",
    phone: "19292439373",
    body: "https://cdn-dev.4jawaly.com/spam-regulation.pdf",
    quotedMsgId: "",
    filename: "filename.pdf"
};

if (!params.token) {
    console.log("token: It must not be a null value");
    process.exit();
}

if (!params.phone) {
    console.log("phone: It must not be a null value");
    process.exit();
}

if (!params.body) {
    console.log("body: It must not be a null value");
    process.exit();
}

const url = `${domain}api/v1/message/file`;
const headers = { Accept: "application/json" };

axios.post(url, params, { headers })
    .then(response => {
        const result = response.data;
        const code = result.code;
        const message = response.status === 200 ? "تم الارسال بنجاح" : result.message;
        const id = result.id;

        console.log(`code: ${code}`);
        console.log(`message: ${message}`);
        console.log(`id: ${id}`);
    })
    .catch(error => {
        if (error.response) {
            const result = error.response.data;
            const code = result.code;
            const message = result.message;

            console.log(`code: ${code}`);
            console.log(`message: ${message}`);
        } else {
            console.log(error.message);
        }
    });
